tests/functional/gcs/endpoint/test_set_subscription_id.py

```diff
@@ -0,0 +1,64 @@
+import uuid
+
+import pytest
+import responses
+from globus_sdk._testing import load_response_set
+
+
+@pytest.mark.parametrize("subscription_id", (str(uuid.uuid4()), "DEFAULT", "null"))
+def test_gcs_endpoint_set_subscription_id(subscription_id, run_line, add_gcs_login):
+    meta = load_response_set("cli.collection_operations").metadata
+    endpoint_id = meta["endpoint_id"]
+    gcs_hostname = meta["gcs_hostname"]
+    add_gcs_login(endpoint_id)
+
+    responses.put(
+        f"https://{gcs_hostname}/api/endpoint/subscription_id",
+        json={
+            "DATA_TYPE": "result#1.0.0",
+            "code": "success",
+            "detail": "success",
+            "has_next_page": False,
+            "http_response_code": 200,
+            "message": f"Updated Endpoint {endpoint_id}",
+        },
+    )
+
+    result = run_line(
+        f"globus gcs endpoint set-subscription-id {endpoint_id} {subscription_id}"
+    )
+
+    assert f"Updated Endpoint {endpoint_id}" in result.stdout
+
+
+def test_gcs_endpoint_set_subscription_id__when_not_subscription_manager(
```

Comment:
```suggestion
def test_gcs_endpoint_set_subscription_id_when_not_subscription_manager(
```

Reply:
This extra _ was intentional.

It's a test naming convention of
```
def test_operation[__when_condition][__then_result]
```

Reply:
I wondered if it was intentional. Thanks for sharing that convention!

Reply:
Yep, one of those little naming tricks that helps frame the way you think about writing a test.